### PR TITLE
Remove AddGeneratorFlag from kubectl create deployment

### LINKED ISSUE #99897: Post generators flags cleanup
So with #99733 and #99732 we are almost done cleaning deprecations.

There are some follow ups that need to be done:

* [ ] In staging/src/k8s.io/kubectl/pkg/cmd/autoscale.go the deprecated flag "generator" must be removed - @MadhavJivrajani 
* [ ] In staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go the call to AddGeneratorFlag must be replaced to 	cmdutil.AddDryRunFlag(cmd) and then in staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go the function AddGeneratorFlags can be removed - @BLasan

This is a good first issue. If you already did any contribution to Kubernetes, **please leave this issue for someone that never contributed!**
If you are a first time contributor, please comment on this issue with what item you wanna work, and I'll assign it to you. Then when the PR is ready, please mark me (/assign @rikatz and /cc @rikatz) and I'll review it for you!

/sig cli
/kind cleanup
/triage accepted
/priority backlog
/good-first-issue

### PR DESCRIPTION
**Pull request by BLasan**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR replaces the `AddGenratorFlag `function with `AddDryRunFlag` function used in `create_deployment.go` file. And removed the unwanted `AddGenratorFlag` function from utils file

#### Which issue(s) this PR fixes:
Partial fix for #99897

```release-note
Remove the deprecated flag --generator from kubectl create deployment command
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

/assign @rikatz 
/cc @rikatz

### CODE REVIEW

**Comment by rikatz** [comment]
/retest

I guess this one is good to go:

* AddGeneratorFlags is used only by kubectl create deployment and should actually have been removed with that migration
* The PR #88655 deprecated the flag, and it was released with v1.18 (more than one year, two releases behind).

**Comment by rikatz** [comment]
@BLasan on the PR description, at the part of release note, please add something like:

"Remove the deprecated flag --generator from kubectl create deployment command"

**Comment by BLasan** [comment]
> @BLasan on the PR description, at the part of release note, please add something like:
> 
> "Remove the deprecated flag --generator from kubectl create deployment command"

Will add sir. Thanks :)

**Comment by BLasan** [comment]
> #### What type of PR is this?
> /kind cleanup
> 
> #### What this PR does / why we need it:
> This PR replaces the `AddGenratorFlag `function with `AddDryRunFlag` function used in `create_deployment.go` file. And removed the unwanted `AddGenratorFlag` function from utils file
> 
> #### Which issue(s) this PR fixes:
> Fixes #99897 (Partial)
> 
> #### Special notes for your reviewer:
> #### Does this PR introduce a user-facing change?
> No
> 
> #### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
> /assign @rikatz
> /cc @rikatz



> #### What type of PR is this?
> /kind cleanup
> 
> #### What this PR does / why we need it:
> This PR replaces the `AddGenratorFlag `function with `AddDryRunFlag` function used in `create_deployment.go` file. And removed the unwanted `AddGenratorFlag` function from utils file
> 
> #### Which issue(s) this PR fixes:
> Fixes #99897 (Partial)
> 
> #### Special notes for your reviewer:
> #### Does this PR introduce a user-facing change?
> No
>
>  ### Release note
```
Remove the deprecated flag --generator from kubectl create deployment command
```
> #### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
> /assign @rikatz
> /cc @rikatz

**Comment by rikatz** [comment]
you need to edit the PR description :)

**Comment by BLasan** [comment]
> you need to edit the PR description :)

Yuh, edited :)

**Comment by rikatz** [comment]
hum, bot should have removed the label.

Is the description this way? 

image

It's important to be with the release-note identified to bot recognize this :D

**Comment by BLasan** [comment]
> hum, bot should have removed the label.
> 
> Is the description this way?
> 
> image
> 
> It's important to be with the release-note identified to bot recognize this :D

I removed that part from the PR. Will add this sir. Sorry

**Comment by rikatz** [comment]
No need to be sorry :D 

This is the idea of a good first issue, to learn more about how PRs works on Kubernetes :)

**Comment by BLasan** [comment]
> No need to be sorry :D
> 
> This is the idea of a good first issue, to learn more about how PRs works on Kubernetes :)

Thanks sir :)

**Comment by rikatz** [comment]
/triage accepted
/priority backlog
/lgtm

Will leave the final decision to sig-cli approvers, if this should enter on v1.21 or if we should wait until v1.22 branch is opened :)

Thank you for your contribution!

**Comment by rikatz** [comment]
Oh btw some more small nits:

Please change in the description from Fixes #lalal to "Partial fix for #99897", otherwise the original issue will be closed as soon as this gets merged.

And also, still in the "Does this PR introduce a user-facing change?" section there's a leading "No" there :) This needs to be removed, as what matters is what's inside the release-note block that you've changed

**Comment by BLasan** [comment]
> #### What type of PR is this?
> /kind cleanup
> 
> #### What this PR does / why we need it:
> This PR replaces the `AddGenratorFlag `function with `AddDryRunFlag` function used in `create_deployment.go` file. And removed the unwanted `AddGenratorFlag` function from utils file
> 
> #### Which issue(s) this PR fixes:
> Fixes #99897 (Partial)
> 
> #### Special notes for your reviewer:
> #### Does this PR introduce a user-facing change?
> No
> 
> ```
> Remove the deprecated flag --generator from kubectl create deployment command
> ```
> 
> #### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
> /assign @rikatz
> /cc @rikatz




> Oh btw some more small nits:
> 
> Please change in the description from Fixes #lalal to "Partial fix for #99897", otherwise the original issue will be closed as soon as this gets merged.
> 
> And also, still in the "Does this PR introduce a user-facing change?" section there's a leading "No" there :) This needs to be removed, as what matters is what's inside the release-note block that you've changed

Looks good sir?

**Comment by BLasan** [comment]
> Oh btw some more small nits:
> 
> Please change in the description from Fixes #lalal to "Partial fix for #99897", otherwise the original issue will be closed as soon as this gets merged.
> 
> And also, still in the "Does this PR introduce a user-facing change?" section there's a leading "No" there :) This needs to be removed, as what matters is what's inside the release-note block that you've changed

Looks good sir?

**Comment by rikatz** [comment]
yup :) Thanks!

**Comment by BLasan** [comment]
> yup :) Thanks!

Thanks for the guidance and support :)

**Comment by rikatz** [comment]
/hold

Let's wait until v1.22 cycle starts (we're too close from code freeze right now)

**Comment by BLasan** [comment]
> /hold
> 
> Let's wait until v1.22 cycle starts (we're too close from code freeze right now)

Any updates on this PR?

**Comment by rikatz** [comment]
/retitle Remove AddGeneratorFlag from kubectl create deployment

**Comment by BLasan** [comment]
@rikatz Please review

**Review by rikatz — COMMENTED** [review]
/lgtm

From my side is good, thank you!

/hold

Will wait the final word from @soltysh if this should enter v1.22 per deprecation policy or not :D

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/hold cancel

**Pull request merged**
